### PR TITLE
Add a file documenting how to connect to the MySQL database in a Docker container

### LINKED ISSUE #297: Write instructions for connecting to the MySQL database from the host machine
This is a reminder for myself either later tonight or tomorrow (unless someone beats me to it!).

I figured I'd put this under `docs/database/connecting.md`.

### PR DESCRIPTION
**Pull request by Phrohdoh**
[Rendered](https://github.com/Phrohdoh/monica/blob/3ad7d468c07a68524f5871eba6751a51d8b23a67/docs/database/connecting.md)
---

This is the first draft. As such please review and address the TODOs I left in the markdown.

---

Where should I link to this? From the Docker setup instructions?

---

If merged this will close #297.

### CODE REVIEW

File: docs/database/connecting.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- alter test data
+- generate custom metrics
+- ...
+
+---
+
+<!-- TODO(review): Is there a better way to express/show these file changes? -->
+1. Stop your Monica and MySQL containers if they are running
+1. Bind port 3306 in your container to 3306 on your host machine
+    <!-- TODO(review): Are these changes safe to include directly in-tree? -->
```
Comment (djaiss) [inline docs/database/connecting.md]: I don't have the answer to that. Docker is a mystery to me :-/

File: docs/database/connecting.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Connect to MySQL inside of a Docker container
```
Comment (djaiss) [inline docs/database/connecting.md]: `# Connect to MySQL inside of a Docker container` should become

```markdown
# Connect to MySQL

There may come a time when....

## Inside of a Docker container

1. Stop your Monica...
```

File: docs/database/connecting.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <!-- TODO(review): Are these changes safe to include directly in-tree? -->
+    <!-- .. can we expose 3306 only in development instances somehow? -->
+    1. Add `EXPOSE 3306:3306` to your `Dockerfile` under `EXPOSE 80:80`
+    1. Add a `ports` node under `mysql` in `docker-compose.yml` with a `3306:3306` entry
+1. Start your containers with `docker-compose up`, wait for them to spin up
+1. Connect to the database via a MySQL client (CLI, GUI, etc.)
+
+---
+
+On macOS the free and open-source **[Sequel Pro](http://www.sequelpro.com/)** is a solid choice.
```
Comment (djaiss) [inline docs/database/connecting.md]:
```markdown

## Sequel Pro (macOS)

File: docs/database/connecting.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+would like to connect directly to the MySQL database being used.
+
+For example you may want to
+- alter test data
+- generate custom metrics
+- ...
+
+---
+
+<!-- TODO(review): Is there a better way to express/show these file changes? -->
```
Comment (djaiss) [inline docs/database/connecting.md]: No I think it's good as is.

**Comment by djaiss** [comment]
Thanks for this.

I would link this document from two places:
1. `readme.md` at the bottom of `Setup the testing environment`.
1. `docker.md` somewhere

Thanks

File: docs/database/connecting.md (hunk 1 above)
Comment (Phrohdoh) [inline docs/database/connecting.md]: My understanding is that this should be safe so long as the host machine itself doesn't expose 3306 (unless you _want_ other machines to be able to connect to the db).

File: docs/database/connecting.md (hunk 2 above)
Comment (Phrohdoh) [inline docs/database/connecting.md]: Those steps are performed on the host machine not from inside of the container.

**Comment by Phrohdoh** [comment]
@djaiss I have updated this PR. Please re-review.

If you still want me to make [these changes](https://github.com/monicahq/monica/pull/300#discussion_r122185391) I can.

File: docs/database/connecting.md (hunk 2 above)
Comment (djaiss) [inline docs/database/connecting.md]: Oh right.

**Pull request merged**